Public/Src/Cache/MemoizationStore/Interfaces/Sessions/ContentHashListWithDeterminism.cs

```diff
@@ -20,6 +22,22 @@ public ContentHashListWithDeterminism(ContentHashList contentHashList, CacheDete
             Determinism = determinism;
         }
 
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="ContentHashListWithDeterminism"/> struct from a binary
+        ///     representation
+        /// </summary>
+        public ContentHashListWithDeterminism(BinaryReader reader)

```

Comment:
It seems that this is slighly inconsistent: in most cases we have serialize/deserialize methods, but here we use constructor.

I would prefer having explicit serialization/deserialization methods to be more consistent.